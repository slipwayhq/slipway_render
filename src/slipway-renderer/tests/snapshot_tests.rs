use slipway_renderer::{default_host_config, render, DebugMode};

#[test]
fn snapshots() {
    insta::with_settings!({}, {
        insta::glob!("snapshot_inputs/*.json", |path| {
            let json_data = std::fs::read_to_string(path).unwrap();
            let (image, card) = render(
                &default_host_config(),
                &json_data,
                1000,
                800,
                DebugMode::full(),
            )
            .unwrap();

            image.save(path.with_extension("png")).unwrap();
            insta::assert_json_snapshot!(card);
        });
    });
}
