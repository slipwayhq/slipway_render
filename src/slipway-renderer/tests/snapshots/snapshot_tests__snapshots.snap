---
source: slipway-renderer/tests/snapshot_tests.rs
expression: card
input_file: slipway-renderer/tests/snapshot_inputs/mixed_container_heights.json
---
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "body": [
    {
      "type": "Container",
      "items": [
        {
          "type": "TextBlock",
          "text": "1",
          ".layout": {
            "rect": {
              "x": 24,
              "y": 24,
              "width": 100,
              "height": 10
            },
            "placement": "SoleVertical"
          }
        }
      ],
      "style": "accent",
      ".layout": {
        "rect": {
          "x": 12,
          "y": 12,
          "width": 976,
          "height": 34
        },
        "placement": "Top"
      }
    },
    {
      "type": "Container",
      "height": "stretch",
      "items": [
        {
          "type": "TextBlock",
          "text": "2",
          ".layout": {
            "rect": {
              "x": 24,
              "y": 70,
              "width": 100,
              "height": 10
            },
            "placement": "SoleVertical"
          }
        }
      ],
      "style": "accent",
      ".layout": {
        "rect": {
          "x": 12,
          "y": 58,
          "width": 976,
          "height": 131
        },
        "placement": "WithinVertical"
      }
    },
    {
      "type": "Container",
      "height": "stretch",
      "items": [
        {
          "type": "TextBlock",
          "text": "3",
          ".layout": {
            "rect": {
              "x": 24,
              "y": 213,
              "width": 100,
              "height": 10
            },
            "placement": "SoleVertical"
          }
        }
      ],
      "style": "accent",
      ".layout": {
        "rect": {
          "x": 12,
          "y": 201,
          "width": 976,
          "height": 132
        },
        "placement": "WithinVertical"
      }
    },
    {
      "type": "Container",
      "height": "stretch",
      "items": [
        {
          "type": "TextBlock",
          "text": "4",
          ".layout": {
            "rect": {
              "x": 24,
              "y": 357,
              "width": 100,
              "height": 10
            },
            "placement": "SoleVertical"
          }
        }
      ],
      "minHeight": "300px",
      "style": "accent",
      ".layout": {
        "rect": {
          "x": 12,
          "y": 345,
          "width": 976,
          "height": 397
        },
        "placement": "WithinVertical"
      }
    },
    {
      "type": "Container",
      "items": [
        {
          "type": "TextBlock",
          "text": "5",
          ".layout": {
            "rect": {
              "x": 24,
              "y": 766,
              "width": 100,
              "height": 10
            },
            "placement": "SoleVertical"
          }
        }
      ],
      "style": "accent",
      ".layout": {
        "rect": {
          "x": 12,
          "y": 754,
          "width": 976,
          "height": 34
        },
        "placement": "Bottom"
      }
    }
  ],
  "minHeight": "800px",
  "version": "1.5",
  "type": "AdaptiveCard",
  ".layout": {
    "rect": {
      "x": 0,
      "y": 0,
      "width": 1000,
      "height": 800
    }
  }
}
